src/scripts/Add-Scope.ps1

```diff
@@ -0,0 +1,241 @@
+    $payload = '{
+      "properties": {
+        "schedule": {
+          "status": "Inactive",
+          "recurrence": "daily",
+          "recurrencePeriod": {
+            "from": "2099-06-01T00:00:00Z",
+            "to": "2099-10-31T00:00:00Z"
+          }
+        },
+        "partitionData": "true",
+        "format": "Csv",
+        "deliveryInfo": {
+          "destination": {
+            "resourceId": "{0}",
+            "container": "{1}",
+            "rootFolderPath": "{2}"
+          }
+        },
+        "definition": {
+          "type": "{3}",
+          "timeframe": "Custom",
+          "timePeriod" : {
+            "from" : "{4}",
+            "to" : "{5}"
+          },
+          "dataSet": {
+            "granularity": "Daily"
+          }
+        }
+      }
+    }'
+  
+    $payload = $payload.Replace("{0}", $ApiParams.DestinationResourceId)
+    $payload = $payload.Replace("{1}", $ApiParams.DestinationContainer)
+    $payload = $payload.Replace("{2}", $ApiParams.DestinationRootFolderPath)
+    $payload = $payload.Replace("{3}", $ApiParams.DefinitionType)
+    $payload = $payload.Replace("{4}", $ApiParams.TimePeriodFrom)
+    $payload = $payload.Replace("{5}", $ApiParams.TimePeriodTo)
+    $payload = $payload.Replace("{6}", $ApiParams.PartitionData)
+  }
+  else {
+    $payload = '{
+      "properties": {
+        "schedule": {
+          "status": "{8}",
+          "recurrence": "{7}",
+          "recurrencePeriod": {
+            "from": "{6}",
+            "to": "{9}"
+          }
+        },
+        "partitionData": "{5}",
+        "format": "Csv",
+        "deliveryInfo": {
+          "destination": {
+            "resourceId": "{0}",
+            "container": "{1}",
+            "rootFolderPath": "{2}"
+          }
+        },
+        "definition": {
+          "type": "{3}",
+          "timeframe": "{4}",
+          "dataSet": {
+            "granularity": "Daily"
+          }
+        }
+      }
+    }'
+
+    $payload = $payload.Replace("{0}", $ApiParams.DestinationResourceId)
+    $payload = $payload.Replace("{1}", $ApiParams.DestinationContainer)
+    $payload = $payload.Replace("{2}", $ApiParams.DestinationRootFolderPath)
+    $payload = $payload.Replace("{3}", $ApiParams.DefinitionType)
+    $payload = $payload.Replace("{4}", $ApiParams.DefinitionTimeframe)
+    $payload = $payload.Replace("{5}", $ApiParams.PartitionData)
+    $payload = $payload.Replace("{6}", $ApiParams.RecurrencePeriodFrom)
+    $payload = $payload.Replace("{9}", $ApiParams.RecurrencePeriodTo)
+    $payload = $payload.Replace("{7}", $ApiParams.ScheduleRecurrence)
+    $payload = $payload.Replace("{8}", $ApiParams.ScheduleStatus)
+  }
+  
+  $apiResult = Invoke-AzRestMethod -Uri $uri -Method PUT -Payload $payload
+  if ($apiResult.StatusCode -ne "201") {
+    $apiResult
+    Throw "Cost Management API call failed"
+  }
+}
+
+function Set-CostManagementExport {
```

Comment:
Would it be better for us to move this to bicep? I can also work on getting it into the Bicep Registry. (Looking into that now for another type.)

Reply:
Yes